docs/user/troubleshootingbootstrap.md

```diff
@@ -0,0 +1,219 @@
+# Troubleshooting Bootstrap Failures
+
+Unfortunately, there will always be some cases where OpenShift fails to install properly. In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
+
+## Gathering bootstrap failure logs
+
+### Using the installer provisioned workflow
+
+When users are using the installer to create the OpenShift cluster, the installer has all the information to automatically capture the logs from bootstrap host in case of failure.
+
+#### Authenticating with bootstrap host for ipi
+
+The installer will use the user's environment to discover the credentials to connect to the bootstrap host over SSH. One of the following methods is used by the installer,
+
+1. Use the user's already setup `SSH_AGENT`. If the user has a ssh-agent setup, the installer will use it for SSH authentication.
+
+2. Use the user'd home directory, `~/.ssh` on Linux hosts, to load all the SSH private keys and use those for SSH authentication.
+    a. The installer also configures the bootstrap host with a *generated* SSH key, and this private key will be used for SSH authentication none of the user keys are trusted.
```

Comment:
The placement feels odd... should this be `3.`?

Reply:
this is not 3, this is only valid in case of the discovering keys, if SSH_AGENT is set, we don't do any discovering.